extensions/schema-registry/apicurio/avro/deployment/src/main/java/io/quarkus/apicurio/registry/avro/ApicurioRegistryAvroProcessor.java

```diff
@@ -30,11 +31,16 @@ public void apicurioRegistryAvro(BuildProducer<ReflectiveClassBuildItem> reflect
                 "io.apicurio.registry.serde.avro.strategy.TopicRecordIdStrategy"));
 
         reflectiveClass.produce(new ReflectiveClassBuildItem(true, true, true,
-                "io.apicurio.registry.serde.DefaultSchemaResolver",
                 "io.apicurio.registry.serde.DefaultIdHandler",
                 "io.apicurio.registry.serde.Legacy4ByteIdHandler",
                 "io.apicurio.registry.serde.fallback.DefaultFallbackArtifactProvider",
                 "io.apicurio.registry.serde.headers.DefaultHeadersHandler"));
+
+        String defaultSchemaResolver = "io.apicurio.registry.serde.DefaultSchemaResolver";
+        if (QuarkusClassLoader.isClassPresentAtRuntime(defaultSchemaResolver)) {
+            // Class not present after 2.2.0.Final, the registration may be removed once apicurio 1.x support is dropped
```

Comment:
If the class is present in early Apicurio Registry 2.x libraries, we need to keep this as long as we support early Apicurio Registry 2.x libraries.

```suggestion
            // Class not present after 2.2.0.Final
```

or something like that, I'd guess.

Reply:
OK, I based this on the comment:

> For integration-tests/kafka-avro-apicurio2, I believe we want to use the same version of Apicurio Registry 2.x libraries that we compile against elsewhere.

I will update the comments.